### PR TITLE
R/JS api clients to support POST requests

### LINKED ISSUE #317: Switch Coffee, JS, and R Epidata clients to use POST instead of GET for long queries
We expedited https://github.com/cmu-delphi/delphi-epidata/pull/309, but now the Python Epidata client (yet unreleased) doesn't match the Coffee, JS, and R client behavior.

Use the Python implementation of POST as a template to bring the others back in line.

### PR DESCRIPTION
**Pull request by sgratzl**
closes #317 

are there any automatic test setups which will test that?

### CODE REVIEW

**Comment by dfarrow0** [comment]
> are there any automatic test setups which will test that?

if you're meaning integration tests for js/r clients, then sadly no (at least that i'm aware of)

**Comment by dfarrow0** [comment]
thanks for taking this on! i noticed that you added js/html files to test the js client. how do you run this locally? i'd be happy to add instruction to the dev guide and possibly automate this.

**Comment by sgratzl** [comment]
atm they are simple smoke tests. You open the HTML file in the browser and it should show two success messages. Same for the node script that you just run. We could think of using a proper testing framework like jest or simple based on `console.assert`

File: src/client/delphi_epidata.coffee [hunk 1]
```diff
@@ -42,9 +42,19 @@ class Epidata
       else
         callback(0, 'unknown error', null)
     # Request data from the server
-    if $?.getJSON?
+    isBrowser = $?.ajax?
+    data = if isBrowser then $.param(params) else querystring.stringify(params)
+    fullURL = "#{BASE_URL}?#{data}"
+    method = if fullURL.length < 2048 then 'GET' else 'POST'
```
Comment (krivard) [inline src/client/delphi_epidata.coffee:48]: add a comment here about avoiding 414

File: integrations/client/test_delphi_epidata.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<script src="https://unpkg.com/jquery"></script>
+<script src="../../src/client/delphi_epidata.js"></script>
+<script>
+    function cb(r, msg, data) {
+        console.log(r, msg, data);
+    }
+    Epidata.covidcast(cb, 'fb-survey', Array(1).fill('raw_cli').join(','), 'day', 'state', '2020-12-01', 'ca');
+    Epidata.covidcast(cb, 'fb-survey', Array(1000).fill('raw_cli').join(','), 'day', 'state', '2020-12-01', 'ca');
+</script>
```
Comment (krivard) [inline integrations/client/test_delphi_epidata.html:9]: add newline at EOF

File: integrations/client/test_delphi_epidata.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const Epidata = require('../../src/client/delphi_epidata').Epidata;
+
+function cb(r, msg, data) {
+    console.log(r, msg, data);
+}
+Epidata.covidcast(cb, 'fb-survey', Array(1).fill('raw_cli').join(','), 'day', 'state', '2020-12-01', 'ca');
+Epidata.covidcast(cb, 'fb-survey', Array(1000).fill('raw_cli').join(','), 'day', 'state', '2020-12-01', 'ca');
```
Comment (krivard) [inline integrations/client/test_delphi_epidata.js:7]: add newline at EOF

**Review by krivard — APPROVED** [review]
nice work! 

this lays the groundwork for adding a test runner; something that will output pass or fail with a docker command. out of scope for this PR but I'll make an issue for it.

**Review by krivard — CHANGES_REQUESTED** [review]
[wrong radio button]

File: src/client/delphi_epidata.coffee (hunk 1 above)
Comment (sgratzl) [reply src/client/delphi_epidata.coffee:50]: done

File: integrations/client/test_delphi_epidata.html (hunk 2 above)
Comment (sgratzl) [reply integrations/client/test_delphi_epidata.html]: done

File: integrations/client/test_delphi_epidata.js (hunk 3 above)
Comment (sgratzl) [reply integrations/client/test_delphi_epidata.js]: done

**Review by krivard — APPROVED** [review]

**Pull request merged**